### DISCUSSION TITLE
So what is the safest way to store Directus credentials?

### DISCUSSION THREAD

**Discussion by AbraamNader** [directus/directus#6996] [Archive]
I am not a NodeJS developer (yet), so I a have newbie question:
How safe is having credentials in .env files? can other node modules dependencies within Directus/sql provider read them and phone home?

the only way I could think of was to secure my setup by making database/redis accessible only within the local network.

So what is the safest way to store Directus credentials?

Note:
I am using AWS.

**Comment by rijkvanzanten** [comment]
Most tools/services heavily rely on environment variables for credentials, including Docker. Technically speaking, yes, a nefarious package under node_modules could read them.

> the only way I could think of was to secure my setup by making database/redis accessible only within the local network.

I would always recommend doing this, regardless of what you use to connect to the database. Ideally, you want to prevent public network access from reaching the database at all times. If you must, you can add an allow list of IP addresses to the DBs firewall for local debugging using something like TablePlus, though that should always be temporary.

If you wanna one-up the safety of the keys within AWS itself, I believe you can use AWS Systems Manager Parameter Store when running Directus in AWS Fargate to store / pass along the environment keys safely. That being said, I'm no expert in AWS myself so I can't say anything to the benefits of that particular setup.

**Reply by AbraamNader** [reply]
The whole NodeJS environment is still new to me, I am still learning NodeJS, coming from .NET, where we would encrypt sensitive configurations, and the .NET app decrypts it in memory when it runs, that is why I am asking, just to make sure I'm not doing things wrong.

Thank you so much for your reply.
